### PR TITLE
Spring-clean of golangci-lint config

### PR DESCRIPTION
**Pull request by peterebden**
Enabled a few new friends which do some mildly nice things

### CODE REVIEW

File: src/test/coverage_test.go [hunk 1]
```diff
@@ -157,6 +157,7 @@ func TestParseBlocks(t *testing.T) {
 }
 
 func assertLine(t *testing.T, lines []core.LineCoverage, i int, expected core.LineCoverage) {
+	t.Helper()
```
Comment (Tatskaari) [inline src/test/coverage_test.go:160]: Ah nice! These do help in test output.

File: src/run/run_test.go [hunk 2]
```diff
@@ -37,7 +37,7 @@ func TestEnvVars(t *testing.T) {
 	config.Build.Path = []string{"/wibble"}
 	state, lab1, _ := makeState(config)
 
-	os.Setenv("PATH", "/usr/local/bin:/usr/bin:/bin")
+	t.Setenv("PATH", "/usr/local/bin:/usr/bin:/bin")
```
Comment (Tatskaari) [inline src/run/run_test.go:40]: Oh I didn't know this was a thing.

**Review by Tatskaari — APPROVED** [review]
Nice! Looks like some useful lint output.

File: src/run/run_test.go (hunk 2 above)
Comment (tiagovtristao) [reply src/run/run_test.go:40]: Yah, this is quite neat.

**Review by tiagovtristao — APPROVED** [review]

**Pull request merged**